tests/publiccloud/migration.pm

```diff
@@ -66,11 +66,14 @@ sub run {
     # migration finished and instance rebooted
     record_info('INFO', 'Checking the migration succeed');
 
-    my $product_version = $instance->ssh_script_run(cmd => 'sudo cat /etc/os-release');
+    my $product_version = $instance->ssh_script_run(cmd => 'sudo cat /etc/os-release | grep ^VERSION_ID');
     record_info('INFO', $product_version);
 
     my $migrated_version = 'N/A';
-    $migrated_version = $1 if $product_version =~ /^VERSION_ID="([\d\.]+)"/sm;
+    my $get_version_id_cmd = "grep '^VERSION_ID=' /etc/os-release | cut -d'=' -f2 | tr -d '\"'";
+    $migrated_version = $instance->run_ssh_command(cmd => $get_version_id_cmd);
```

Comment:
no need to do a second call to get what you already have in `$product_version` variable . please stick to old approach and extract all needed info from a variable using perl .  Also you can find out right regex without running dozens of tests . just copy content of os release to local file and create small perl file which trying to grep it

Reply:
As you suggested tried with small perl file and fixed the code back in perl